src/openrct2-ui/windows/Guest.cpp

```diff
@@ -1942,58 +1947,67 @@ static rct_string_id window_guest_inventory_format_item(Peep* peep, int32_t item
     auto parkName = park.Name.c_str();
 
     // Default arguments
-    set_format_arg(0, uint32_t, ShopItems[item].Image);
-    set_format_arg(4, rct_string_id, ShopItems[item].Naming.Display);
-    set_format_arg(6, rct_string_id, STR_STRING);
-    set_format_arg(8, const char*, parkName);
+    auto ft = Formatter::Common();
+    ft.Add<uint32_t>(ShopItems[item].Image);
+    ft.Add<rct_string_id>(ShopItems[item].Naming.Display);
+    ft.Add<rct_string_id>(STR_STRING);
+    ft.Add<const char*>(parkName);
 
     // Special overrides
     Ride* ride{};
     switch (item)
     {
         case SHOP_ITEM_BALLOON:
-            set_format_arg(0, uint32_t, SPRITE_ID_PALETTE_COLOUR_1(peep->BalloonColour) | ShopItems[item].Image);
+            ft.Increment(-ft.NumBytes());
```

Comment:
Argh this is very hackish X_X
I'm going to think if there's a better way. Seems like here, and on other `case`s you're just trying to rewrite part of what you wrote before the `switch`. I'd save each of these arguments:
```
    ft.Add<uint32_t>(ShopItems[item].Image);
    ft.Add<rct_string_id>(ShopItems[item].Naming.Display);
    ft.Add<rct_string_id>(STR_STRING);
    ft.Add<const char*>(parkName);
```
in a corresponding variable, and update them on the cases, and then actually call:
```
    ft.Add<uint32_t>(ShopItems[item].Image);
    ft.Add<rct_string_id>(ShopItems[item].Naming.Display);
    ft.Add<rct_string_id>(STR_STRING);
    ft.Add<const char*>(parkName);
```
after the switch to get the properly formatted message with decrementing

Reply:
This had already been changed when you wrote this...

Reply:
I did review it commit by commit, maybe that's why